codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -91,6 +91,11 @@
      */
     private ByteBuf buffer;
 
+    /**
+     * Promise associated with writing out the contents of {@link #buffer}.
+     */
+    private ChannelPromise bufferPromise;
```

Comment:
You will also need to do something with this `bufferPromise` in `handlerRemoved(...)`. Otherwise you may end up in a situation of this promise never been notified.

Reply:
++ right, I made it cancel the promise in this case now.

Reply:
I think it would be better to fail it with an exception

Reply:
++ made it fail with exception now